backend/handlers.go

```diff
@@ -49,7 +59,33 @@ func (cs *server) publishHandler(w http.ResponseWriter, r *http.Request, _ httpr
 	}
 	defer body.Close()
 
-	cs.publish(msg)
+	// Unmarshal the JSON message into a map
+	var data map[string]interface{}
+	if err := json.Unmarshal(msg, &data); err != nil {
```

Comment:
Create a decoder and pass the body to the decoder. This time you have to allocate less memory.